geopandas/geodataframe.py

```diff
@@ -297,6 +409,12 @@ def fill_none(row):
 
             yield feature
 
+    def iterrows(self):
```

Comment:
Note for myself: we will need some machinery (eg decorator, maybe can use from pandas) to add the docstring back of the parent method, if we only change some implementation to get it working correctly (so if we don't need a custom docstrin)